tools/setup/install-shellcheck

```diff
@@ -2,8 +2,15 @@
 set -eu
 
 version=0.7.1
-tarball="shellcheck-v$version.linux.x86_64.tar.xz"
-sha256=64f17152d96d7ec261ad3086ed42d18232fcb65148b44571b564d688269d36c8
+if [[ $(uname -m) == "aarch64" ]]; then
+    # ARM 64bit
+    tarball="shellcheck-v$version.linux.aarch64.tar.xz"
+    sha256=b50cc31509b354ab5bbfc160bc0967567ed98cd9308fd43f38551b36cccc4446
+else
+    # Assume x86 64bit
+    tarball="shellcheck-v$version.linux.x86_64.tar.xz"
+    sha256=64f17152d96d7ec261ad3086ed42d18232fcb65148b44571b564d688269d36c8
+fi
```

Comment:
I’d propose

```bash
arch="$(uname -m)"
tarball="shellcheck-v$version.linux.$arch.tar.xz"
declare -A sha256=(
    [aarch64]=b50cc31509b354ab5bbfc160bc0967567ed98cd9308fd43f38551b36cccc4446
    [x86_64]=64f17152d96d7ec261ad3086ed42d18232fcb65148b44571b564d688269d36c8
)
```

and then below use `${sha256[$arch]}`, which will automatically fail if the architecture isn’t recognized.

Reply:
Yeah that's nicer.

Not that familiar with bash, didn't know you can have a map construct.